test/functional/currency/currency-parser.js

```diff
@@ -0,0 +1,146 @@
+define([
+	"globalize",
+	"json!cldr-data/main/de/currencies.json",
+	"json!cldr-data/main/de/numbers.json",
+	"json!cldr-data/main/en/currencies.json",
+	"json!cldr-data/main/en/numbers.json",
+	"json!cldr-data/main/zh/currencies.json",
+	"json!cldr-data/main/zh/numbers.json",
+	"json!cldr-data/supplemental/currencyData.json",
+	"json!cldr-data/supplemental/likelySubtags.json",
+	"json!cldr-data/supplemental/plurals.json",
+	"../../util",
+
+	"globalize/currency",
+	"globalize/number",
+	"globalize/plural"
+], function( Globalize, deCurrencies, deNumbers, enCurrencies, enNumbers, zhCurrencies,
+	zhNumbers, currencyData, likelySubtags, plurals, util ) {
+
+var teslaS = 69900;
+
+function extraSetup() {
+	Globalize.load(
+		currencyData,
+		deCurrencies,
+		deNumbers,
+		enCurrencies,
+		enNumbers,
+		plurals,
+		zhCurrencies,
+		zhNumbers
+	);
+
+}
+
+QUnit.module( ".currencyParser( [options] )", {
+	setup: function() {
+		Globalize.load( likelySubtags, {
+			main: {
+				en: {},
+				"en-IN": {},
+				"tr-CY": {}
+			}
+		});
+		Globalize.locale( "en" );
+	},
+	teardown: util.resetCldrContent
+});
+
+QUnit.test( "should validate parameters", function( assert ) {
+	util.assertParameterPresence( assert, "currency", function() {
+		Globalize.currencyParser();
+	});
+
+	util.assertCurrencyParameter( assert, "currency", function( invalidValue ) {
+		return function() {
+			Globalize.currencyParser( invalidValue );
+		};
+	});
+
+	util.assertPlainObjectParameter( assert, "options", function( invalidValue ) {
+		return function() {
+			Globalize.currencyParser( invalidValue );
+		};
+	});
+});
+
+QUnit.test( "should validate CLDR content", function( assert ) {
+	util.assertCldrContent( assert, function() {
+		Globalize.currencyParser();
+	});
+});
+
+QUnit.test( "should return a currency formatter", function( assert ) {
+	var de, zh;
+
+	extraSetup();
+
+	de = Globalize( "de" );
+	zh = Globalize( "zh" );
+
+	assert.equal( Globalize.currencyParser()( "$69,900.00" ), teslaS );
+	assert.equal( de.currencyParser()( "69.900,00 $" ), teslaS );
+	assert.equal( zh.currencyParser()( "US$69,900.00" ), teslaS );
+
+	assert.equal( Globalize.currencyParser()( "-$69,900.00" ), -teslaS );
+	assert.equal( de.currencyParser()( "-69.900,00 $" ), -teslaS );
+	assert.equal( zh.currencyParser()( "-US$69,900.00" ), -teslaS );
+
+	assert.equal( Globalize.currencyParser()( "69,900.00 USD" ), teslaS );
+	assert.equal( de.currencyParser()( "69.900,00 USD" ), teslaS );
+	assert.equal( zh.currencyParser()( "69,900.00USD" ), teslaS );
+
+	assert.equal( Globalize.currencyParser()( "69,900.00 US dollars" ), teslaS );
+	assert.equal( de.currencyParser()( "69.900,00 US-Dollar" ), teslaS );
+	assert.equal( zh.currencyParser()( "69,900.00美元" ), teslaS );
+
+	assert.equal( Globalize.currencyParser()( "($1.00)" ), -1 );
```

Comment:
We need to include a test using a different numbering system, e.g., arabic.

Reply:
Should this reflect in NumberParser too?

Reply:
Sorry, I didn't follow your question

Reply:
if this is dependent on `NumberParser`, should `NumberParser` also have tests in different numbering systems?

Reply:
Thanks for the clarification. Yes, it does in unit tests. Example https://github.com/globalizejs/globalize/blob/daf1651b5b6ccfa9e62b4de025c61d116431ffb6/test/unit/number/parse.js#L121